photon-ml/src/test/scala/com/linkedin/photon/ml/optimization/LogisticRegressionOptimizationProblemTest.scala

```diff
@@ -14,24 +14,70 @@
  */
 package com.linkedin.photon.ml.optimization
 
-import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
-import com.linkedin.photon.ml.model.Coefficients
-import com.linkedin.photon.ml.normalization.NormalizationContext
-import com.linkedin.photon.ml.optimization.game.GLMOptimizationConfiguration
-import com.linkedin.photon.ml.test.CommonTestUtils
+import java.util.Random
 
+import breeze.linalg.{DenseVector, Vector}
 import org.apache.spark.rdd.RDD
 import org.mockito.Mockito._
 import org.testng.Assert._
-import org.testng.annotations.Test
+import org.testng.annotations.{DataProvider, Test}
 
-class LogisticRegressionOptimizationProblemTest {
-  import LogisticRegressionOptimizationProblemTest._
+import com.linkedin.photon.ml.constants.MathConst
+import com.linkedin.photon.ml.data.{LabeledPoint, SimpleObjectProvider}
+import com.linkedin.photon.ml.model.Coefficients
+import com.linkedin.photon.ml.normalization.NormalizationContext
+import com.linkedin.photon.ml.optimization.game.GLMOptimizationConfiguration
+import com.linkedin.photon.ml.test.{CommonTestUtils, SparkTestUtils}
+
+class LogisticRegressionOptimizationProblemTest extends SparkTestUtils {
   import CommonTestUtils._
+  import LogisticRegressionOptimizationProblemTest._
+
+  def sigmoid(z: Double): Double = 1.0 / (1.0 + math.exp(-z))
```

Comment:
This function is already available in the ```breeze.numerics``` package, but it's so simple I'm not sure it's a huge deal to repeat it here.